pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/formats/protobufnative/PulsarProtobufNativeFormatFactory.java

```diff
@@ -105,17 +105,34 @@ public ChangelogMode getChangelogMode() {
     }
 
     private String extractTopicName(DynamicTableFactory.Context context) {
-
         validateTopic(context.getCatalogTable().getOptions());
-        String topic = context.getCatalogTable().getOptions().get(TOPIC.key());
-        // Maybe catalog mode
-        if (StringUtils.isNullOrWhitespaceOnly(topic)) {
-            final ObjectIdentifier table = context.getObjectIdentifier();
-            topic = TopicName.get(table.getDatabaseName() + "/" + table.getObjectName()).toString();
+
+        String topic;
+        if (isGenericTable(context)) {
+            topic = context.getCatalogTable().getOptions().get(TOPIC.key());
+        } else {
+            String database = context.getObjectIdentifier().getDatabaseName();
+            String objectName = context.getObjectIdentifier().getObjectName();
+            topic = TopicName.get(database + "/" + objectName).toString();
```

Comment:
you can reference https://github.com/streamnative/pulsar-flink/blob/master/pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/catalog/util/PulsarCatalogSupport.java#L284 to generate a complete topic name